core/src/main/java/fr/sncf/osrd/envelope_sim/allowances/MarecoAllowance.java

```diff
@@ -33,19 +34,22 @@
     // potential speed limit under which the train would use too much capacity
     public final double capacitySpeedLimit;
 
+    public final int allowanceIndex;
+
```

Comment:
that's a bad idea. I would feel much better if location metadata was instead added to exceptions once thrown